lib/chainer/functions/connection/embed_id.rb

```diff
@@ -0,0 +1,50 @@
+module Chainer
+  module Functions
+    module Connection
+      class EmbedIDFunction < Chainer::Function
+        def initialize(ignore_label: nil)
+          @ignore_label = ignore_label
+        end
+
+        def self.embed_id(x, w, ignore_label: nil)
+          self.new(ignore_label: ignore_label).(x, w)
+        end
+
+        def forward_cpu(inputs)
+          (x, w) = inputs
+
+          unless @ignore_label
+            return [Chainer::Utils::Array.take(w, x.to_a, axis: 0)]
+          end
+
+          mask = Numo::Bit.cast(x.flatten.to_a.map{|i| i == @ignore_label}).reshape(*x.shape)
+          if mask.where.empty?
+            return [Chainer::Utils::Array.take(w, x.to_a, axis: 0)]
+          end
+          x = x.dup
+          x[mask.where] = 0
+          y = Chainer::Utils::Array.take(w, x.to_a, axis: 0).dup
+
+          ndindex = Chainer::Utils::Array.ndindex(y.shape.take(y.shape.size - 1))
+          mask.where.each {|i| y[*ndindex[i], true] = 0 }
+
+          [y]
+        end
+
+        def backward_cpu(inputs, grad_outputs)
```

Comment:
I am glad if you can support `Cumo` together.

Codes would become like:


```
def backward(inputs, grad_outputs)
  xm = Chainer.get_array_module(*(inputs + grad_outputs))
```

and replace `Numo` with `xm`